scanpy/tests/test_normalization.py

```diff
@@ -56,3 +62,289 @@ def test_normalize_total_view(typ, dtype):
 
     assert not v.is_view
     assert_equal(adata, v)
+
+
+@pytest.mark.parametrize(
+    'sparsity_func', [csr_matrix.toarray, csr_matrix], ids=lambda x: x.__name__
+)
+@pytest.mark.parametrize('dtype', ['float32', 'int64'])
+def test_normalize_pearson_residuals_inputchecks(sparsity_func, dtype):
+
+    adata = _prepare_pbmc_testdata(sparsity_func, dtype)
+
+    # depending on check_values, warnings should be raised for non-integer data
+    if dtype == 'float32':
+
+        adata_noninteger = adata.copy()
+        x, y = np.nonzero(adata_noninteger.X)
+        adata_noninteger.X[x[0], y[0]] = 0.5
+
+        with pytest.warns(UserWarning) as record:
+            sc.experimental.pp.normalize_pearson_residuals(
+                adata_noninteger.copy(), check_values=True
+            )
+        assert len(record) == 1
+        assert "expects raw count data" in record[0].message.args[0]
+
+        with pytest.warns(None) as record:
+            sc.experimental.pp.normalize_pearson_residuals(
+                adata_noninteger.copy(), check_values=False
+            )
+        assert len(record) == 0
+
+    # errors should be raised for invalid theta values
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.normalize_pearson_residuals(adata.copy(), theta=0)
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.normalize_pearson_residuals(adata.copy(), theta=-1)
+
+    # error should be raised for invalid clipping values
+    with pytest.raises(ValueError) as record:
+        sc.experimental.pp.normalize_pearson_residuals(adata.copy(), clip=-1)
```

Comment:
Could these be a bit more specific about the error they are expecting to see? Basically, just add a match statement that makes sure you are telling the user about which variable is off.

Reply:
yes added!